kernel/src/path.rs

```diff
@@ -21,6 +21,8 @@ const UUID_PART_LEN: usize = 36;
 #[internal_api]
 pub(crate) enum LogPathFileType {
     Commit,
+    /// Staged commits are commits with UUID filenames, stored in _delta_log/_staged_commits dir.
+    StagedCommit,
```

Comment:
is a log_tail allowed to have StagedCommits?
```
    // check log_tail is only commits
    // note that LogSegment checks no gaps/duplicates so we don't duplicate that here
    debug_assert!(
        log_tail
            .iter()
            .all(|entry| matches!(entry.file_type, LogPathFileType::Commit)),
        "log_tail should only contain commits"
    );
```

if so, can we update list_log_files and add a test?